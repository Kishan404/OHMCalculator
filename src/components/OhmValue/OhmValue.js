import React, { Component } from 'react';
import PropTypes from 'prop-types';
//import './OhmValue.css';

const OhmValue =(props) => {
    console.log((props.resistance-(props.resistance * props.tolerance)));
    console.log((props.resistance * props.tolerance));
    console.log(props.tolerance);
    console.log(props.resistance);

    return (
        <div className="result">
            <div className="resultrow">
                <div className="resultlable"><label className="label">Resistance:</label></div>
                <div className="value resultvalue"><label >{props.resistance}</label></div>
                <div className="resultunits"><label >ohms</label></div>
            </div>
            <div className="resultrow">
                <div className="resultlable"><label className="label">Tolerance:</label></div>
                <div className="value resultvalue"><label >{(props.tolerance ||0) ? 'Â± '+props.tolerance+'%':'0'}</label></div>
                <div className="resultunits"><label></label></div>
            </div>
            <div className="resultrow">
                <div className="resultlable"><label className="label">Minimum:</label></div>
                <div className="value resultvalue"><label>{props.tolerance? Math.round((props.resistance-Math.round(props.resistance * props.tolerance/100*100000)/100000)*100000)/100000:props.resistance}</label></div>
                <div className="resultunits"><label>ohms</label></div>
            </div>
            <div className="resultrow">
                <div className="resultlable"><label className="label">Maximum:</label></div>
                <div className="value resultvalue"><label >{props.tolerance? Math.round((props.resistance+Math.round(props.resistance * props.tolerance/100*100000)/100000)*100000)/100000:props.resistance}</label></div>
                {/* <div><label className="resultvalue">{props.tolerance? (props.resistance+(props.resistance * props.tolerance/100)):props.resistance}</label></div> */}
                <div className="resultunits"><label >ohms</label></div>
            </div>
            
        </div>
    );
}
OhmValue.propTypes = {
    resistance: PropTypes.number,
    tolerance: PropTypes.number,
    error: PropTypes.string
}
 
export default OhmValue;