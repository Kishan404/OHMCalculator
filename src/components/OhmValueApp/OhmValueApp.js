import React, { Component } from 'react';
import PropTypes from 'prop-types';
import logo from '../../resources/resister.svg';
import './OhmApp.css';
import OhmValueCalculator from '../../OhmValueCalculator';
import ColorBands from '../ColorBands/ColorBands';


class OhmValueApp extends Component {
    constructor(props) {
        super(props);
        this.state = {
            firstDigitColor: props.firstDigitColor,
            secondDigitColor: props.secondDigitColor,
            multiplierColor: props.multiplierColor,
            toleranceColor: props.toleranceColor,
            resistance: props.resistance,
            tolerance: props.tolerance,
        };
        this.handleColorPick = this.handleColorPick.bind(this);
    }

componentDidMount() {

    let ohmcal = new OhmValueCalculator();
    let result = ohmcal.calculateOhmValue(this.props.firstDigitColor,this.props.secondDigitColor,this.props.multiplierColor,this.props.toleranceColor);
    console.log(result);
    this.setState(() => {
        return {
            resistance: result.Ohm,
            tolerance: result.tolerance,
        }
    });

}    
handleColorPick(newcolor,bandType) {

    this.setState(()=> {
        return {
            [bandType]: newcolor
        }
    });

    let ohmcal = new OhmValueCalculator();
    
    debugger;
    this.setState((prevState)=> {
        let result = ohmcal.calculateOhmValue(prevState.firstDigitColor,prevState.secondDigitColor,prevState.multiplierColor,prevState.toleranceColor);
       return {
            resistance: result.Ohm,
            tolerance: result.tolerance
        }
    });

  }


  render() {
    return (
      <div className="App">
        <img src={logo} className="App-logo" alt="logo" />
        <header className="App-header">
              
        </header>
        <h1 className="App-title">{this.props.hearder}</h1>
        
        <p className="App-intro">
          {this.props.subTitle}
          <br />
          Please select colors bands to calculate resitance value. Use the matching image pattern of your resister for correct input colors selection.  
        </p>
        <br/>
        <div>
          <div><ColorBands 
                bandsList = {this.props.bandsList}
                handleColorPick = {this.handleColorPick}
                /></div>
          <div></div>
        </div>

        <p><b>Resistance : {(this.state.resistance+' Ohm '+(this.state.tolerance ? 'Â± '+this.state.tolerance+'%':''))} </b></p>
        
      </div>
    );
  }
}

OhmValueApp.propTypes = {
    bandsList: PropTypes.array,
    hearder : PropTypes.string,
    subTitle : PropTypes.string,
    firstDigitColor: PropTypes.string,
    secondDigitColor: PropTypes.string,
    multiplierColor: PropTypes.string,
    toleranceColor: PropTypes.string,
};

export default OhmValueApp;
