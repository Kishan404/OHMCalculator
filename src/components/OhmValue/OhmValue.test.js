import React, { Component } from 'react';
const ReactTestRenderer = require('react-test-renderer');
import OhmValue from './OhmValue';
import OhmValueApp from '../OhmValueApp/OhmValueApp';

describe('OhmValue result Snapshot test', () => {
    it('Should compare the component with a snapshot', () => {
      
      const component = ReactTestRenderer.create(
        <OhmValue
            resistance = {12}
            tolerance = {0.05}
            error = {undefined}            
        />
      );
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    })
  });

