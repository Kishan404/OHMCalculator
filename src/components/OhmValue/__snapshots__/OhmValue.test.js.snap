// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OhmValue result Snapshot test Should compare the component with a snapshot 1`] = `
<div
  className="result"
>
  <div
    className="result-error"
  />
  <div
    className="resultrow"
  >
    <div
      className="resultlable"
    >
      <label
        className="label"
      >
        Resistance:
      </label>
    </div>
    <div
      className="value resultvalue"
    >
      <label>
        12
      </label>
    </div>
    <div
      className="resultunits"
    >
      <label>
        ohms
      </label>
    </div>
  </div>
  <div
    className="resultrow"
  >
    <div
      className="resultlable"
    >
      <label
        className="label"
      >
        Tolerance:
      </label>
    </div>
    <div
      className="value resultvalue"
    >
      <label>
        Â± 0.05%
      </label>
    </div>
    <div
      className="resultunits"
    >
      <label />
    </div>
  </div>
  <div
    className="resultrow"
  >
    <div
      className="resultlable"
    >
      <label
        className="label"
      >
        Minimum:
      </label>
    </div>
    <div
      className="value resultvalue"
    >
      <label>
        11.994
      </label>
    </div>
    <div
      className="resultunits"
    >
      <label>
        ohms
      </label>
    </div>
  </div>
  <div
    className="resultrow"
  >
    <div
      className="resultlable"
    >
      <label
        className="label"
      >
        Maximum:
      </label>
    </div>
    <div
      className="value resultvalue"
    >
      <label>
        12.006
      </label>
    </div>
    <div
      className="resultunits"
    >
      <label>
        ohms
      </label>
    </div>
  </div>
</div>
`;
